test/org/opensolaris/opengrok/analysis/executables/JarAnalyzerFactoryTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2018, Chris Fraire <cfraire@me.com>.
+ */
+
+package org.opensolaris.opengrok.analysis.executables;
+
+import java.io.IOException;
+import java.io.InputStream;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertSame;
+import org.junit.Test;
+import org.opensolaris.opengrok.analysis.AnalyzerGuru;
+import org.opensolaris.opengrok.analysis.FileAnalyzerFactory;
+
+/**
+ * Represents a container for tests of {@link JarAnalyzerFactory}.
+ */
+public class JarAnalyzerFactoryTest {
+
+    /**
+     * Tests a JAR file.
+     * @throws IOException
+     */
+    @Test
+    public void testJarWrtAnalyzerGuru() throws IOException {
+        InputStream res = getClass().getClassLoader().getResourceAsStream(
+            "org/opensolaris/opengrok/analysis/executables/javajar.bin");
```

Comment:
there is already a jar file in `testdata/sources/java/main.jar`

Reply:
I did copy an existing OpenGrok jar. I thought it was OK to have a test file close at hand like all the other tests which use embedded resources, and I thought it OK to indicate via filename that it is not being matched by name.

Reply:
I get the point about name matching (even though it can be renamed while the test is running) however `opengrok.jar` weighs some 2 megs which is probably too much for matching first couple of bytes.

Reply:
:) I picked a 1000-byte jar.

Reply:
sounds better. I am sure people still behind ADSL will appreciate it :-)